### ISSUE TITLE
Inject instance of a generated class

### ISSUE THREAD

**Issue by rafakob** [google/dagger#321]
Hello,
Is it possible to inject an instance of a class that was generated with annotation processor?
I've created ```MyClass``` using this tutorial -> http://brianattwell.com/android-annotation-processing-pojo-string-generator/ (I've added 'compiler' and 'api' modules to my app project).

I'm providing it in ```@PerApp``` in ```NetworkModule```:
```java
@Provides
@PerApp
MyClass providesMyClass(){
    return new MyClass();
}
```
I've tested field injection and it's working fine:
```java
public class LoginActivity extends AppCompatActivity {
    @Inject LoginPresenter presenter;
    @Inject MyClass myClass; // no error
    ...
}
````

However when I try constructor injection into my presenter, compilation crashes:
```java
// LoginModule.java
@Provides
@PerActivity
LoginPresenter providesLoginActivityPresenter(MyClass myClass) {
    return new LoginPresenter(myClass);
}
```
```Error:(40, 24) error: cannot find symbol variable DaggerAppComponent```
```java
public class App extends Application {
    private AppComponent appComponent;
    private void initAppComponent() {
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .networkModule(new NetworkModule())
                .build();
    }
```

Am I doing something wrong?

**Comment by ronshapiro** [comment]
(fyi this is probably more fitting as a stack overflow question with the dagger-2 tag)

There's probably more error output, do you see anything else?

**Comment by rafakob** [comment]
Yea, I thought about SO, but the thing is that there's not much very information about that issue. All I get is error in Messages:
```
/Users/rafakob/AndroidStudioProjects/MyProject/app/src/main/java/com/rafakob/MyProject/App.java
Error:(40, 24) error: cannot find symbol variable DaggerAppComponent
Error:Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```
First I thought that when Dagger is generating it's own classes, it doesn't know about my generated class. But the field injection is working, so I'm confused now.

**Comment by ronshapiro** [comment]
This should have been fixed, try upgrading to Dagger 2.6 and if it's still an issue, reopen.
